--- gcc/config.gcc.orig	2021-05-04 14:25:34.000000000 -0700
+++ gcc/config.gcc	2021-06-15 16:35:08.000000000 -0700
@@ -702,6 +702,11 @@
       *-*-darwin[4-9]* | *-*-darwin1[0-9]*)
         macos_min=`expr $darwin_maj - 4`
         ;;
+      *-*-darwin21*)
+        # Darwin 20 corresponds to macOS 11.
+        macos_maj=12
+        def_ld64=750
+        ;;
       *-*-darwin20*)
         # Darwin 20 corresponds to macOS 11.
         macos_maj=11
