--- configure.orig	2020-03-09 23:11:12.000000000 -0700
+++ configure	2020-06-28 22:56:57.000000000 -0700
@@ -3181,6 +3181,8 @@
 			;;
 		esac
 
+		FRAMEWORKINSTALLAPPSPREFIX="@@APPLICATIONS_DIR@@"
+
 		prefix=$PYTHONFRAMEWORKINSTALLDIR/Versions/$VERSION
 
 		# Add files for Mac specific code to the list of output
@@ -3374,7 +3376,7 @@
   # has no effect, don't bother defining them
   Darwin/[6789].*)
     define_xopen_source=no;;
-  Darwin/1[0-9].*)
+  Darwin/[1-9][0-9].*)
     define_xopen_source=no;;
   # On AIX 4 and 5.1, mbstate_t is defined only when _XOPEN_SOURCE == 500 but
   # used in wcsnrtombs() and mbsnrtowcs() even if _XOPEN_SOURCE is not defined
